12/umbraco-cms/reference/content-delivery-api/extension-points/extension-api-for-querying.md

```diff
@@ -0,0 +1,308 @@
+
+## Custom sort
+
+Finally, adding a new custom sorting option can enable us to sort our blog posts based on a custom `"publishDate"` Date Picker property. This allows us to arrange the posts in ascending or descending order based on their publication dates.
+
+{% hint style="info" %}
+This sorting should only be used with content nodes that have a published date to ensure accurate results.
+{% endhint %}
+
+To demonstrate this, consider the following implementation example:
+
+{% code title="PublishDateSort.cs" lineNumbers="true" %}
+```csharp
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.DeliveryApi;
+using Umbraco.Cms.Core.Models;
+
+namespace Umbraco.Docs.Samples;
+
+public class PublishDateSort : ISortHandler, IContentIndexHandler
+{
+    private const string SortOptionSpecifier = "publishDate:";
+    private const string FieldName = "publishDate";
+
+    // Querying
+    public bool CanHandle(string query)
+        => query.StartsWith(SortOptionSpecifier, StringComparison.OrdinalIgnoreCase);
+
+    public SortOption BuildSortOption(string sort)
+    {
+        var sortDirection = sort.Substring(SortOptionSpecifier.Length);
+
+        return new SortOption
+        {
+            FieldName = FieldName,
+            Direction = sortDirection.StartsWith("asc", StringComparison.OrdinalIgnoreCase)
+                ? Direction.Ascending
+                : Direction.Descending
+        };
+    }
+
+    // Indexing
+    public IEnumerable<IndexFieldValue> GetFieldValues(IContent content, string? culture)
+    {
+        var publishDate = content.GetValue<DateTime?>("publishDate");
+
+        if (publishDate is null)
+        {
+            return Enumerable.Empty<IndexFieldValue>();
+        }
+
+        return new[]
+        {
+            new IndexFieldValue
+            {
+                FieldName = FieldName,
+                Values = new object[] { publishDate }
+            }
+        };
+    }
+
+    public IEnumerable<IndexField> GetFields()
+        => new[]
+        {
+            new IndexField
+            {
+                FieldName = FieldName,
+                FieldType = FieldType.Date,
+                VariesByCulture = false
+            }
+        };
+}
+```
+{% endcode %}
+
+The provided implementation follows the familiar structure defined by the `IContentIndexHandler` and `ISortHandler` interfaces. And another point to highlight is that we store the `"publishDate"` value in a `Date` format within the index, which requires regenerating the data in the **DeliveryApiContentIndex**.
+
+In the following example request, we also apply the author filter to retrieve only `"blogpost"` content nodes, which we know have the `"publishDate"` field. This helps to ensure that our sorting yields meaningful and relevant results.
```

Comment:
To follow the rest of the article, I think you should lead the "Custom sort" section with the request and the response, before diving into the implementation.

Reply:
There isn't really a pattern for when the example Request and Response should come - in the Selector they are after the implementation and in the Filter section they are before. So I think it is fine to keep the Sort structure as it is.

And good point for repeating the relevant results part